### PR TITLE
fix+test to_timedelta('NaT', box=False)

### LINKED ISSUE #24957: BUG: to_timedelta('NaT', box=False) returns datetime64('NaT')
On pandas 0.23 (correct):
```
In [2]: pd.to_timedelta('NaT', box=False)
Out[2]: numpy.timedelta64('NaT','ns')
```

But on pandas 0.24:
```
In [2]: pd.to_timedelta('NaT', box=False)
Out[2]: numpy.datetime64('NaT')
```

The result is a datetime, not a timedelta.

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #24957
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/scalar/timedelta/test_timedelta.py [hunk 1]
```diff
@@ -309,8 +309,13 @@ def test_iso_conversion(self):
         assert to_timedelta('P0DT0H0M1S') == expected
 
     def test_nat_converters(self):
```
Comment (jreback) [inline pandas/tests/scalar/timedelta/test_timedelta.py]: can you parameterize over box

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/scalar/timedelta/test_timedelta.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: Do you mean over the input to `to_timedelta`?  That would marginally reduce code, whereas parametrizing over `box` would mean completely re-writing this test and probably others.

Comment (jreback) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: for box, you are testing box=False, need to have equivalent for box=True (which we probably already have), but nicer to have in one place.

Comment (jbrockmendel) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: After looking at this a little more, I don't think parametrizing over box makes sense; the code for the relevant assertions isn't shareable.  Makes more sense to collect the related tests to be adjacent in an upcoming test-refactoring pass.

Comment (jreback) [reply pandas/tests/scalar/timedelta/test_timedelta.py]: ok fair

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
actually can you add a whatsnew note in regressions for 0.24.1 (or bug fixes ok too is fine)

**Comment by jbrockmendel** [comment]
Done

**Pull request merged**